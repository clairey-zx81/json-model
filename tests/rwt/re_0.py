# generated from model: ./re_0.model.json
from typing import Callable
import re as re

type Jsonable = None|bool|int|float|str|list[Jsonable]|dict[str, Jsonable]
type CheckFun = Callable[[Jsonable, str], bool]
type PropMap = dict[str, CheckFun]
type TagMap = dict[None|bool|float|int|str, CheckFun]

# regex "/^S/"
jmsc_re_0 = re.compile("^S").search

# define "check_model" ('$')
def check_model(value: Jsonable, path: str = "$") -> bool:
    # '$'
    # "/^S/"
    result = isinstance(value, str) and jmsc_re_0(value) is not None
    return result

