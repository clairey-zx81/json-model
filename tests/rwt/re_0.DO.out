Disassembly of _fun:
              0 COPY_FREE_VARS           4

780           2 RESUME                   0
              4 LOAD_GLOBAL              1 (NULL + isinstance)
             14 LOAD_FAST                0 (v)
             16 LOAD_GLOBAL              2 (str)
             26 CALL                     2
             34 COPY                     1
             36 POP_JUMP_IF_FALSE       10 (to 58)
             38 POP_TOP
             40 PUSH_NULL
             42 LOAD_DEREF               2 (check_re)
             44 LOAD_FAST                0 (v)
             46 CALL                     1
             54 LOAD_CONST               0 (None)
             56 IS_OP                    1
        >>   58 COPY                     1
             60 POP_JUMP_IF_TRUE        22 (to 106)
             62 POP_TOP

781          64 LOAD_DEREF               5 (self)
             66 LOAD_ATTR                5 (NULL|self + _no)
             86 LOAD_DEREF               4 (mpath)
             88 LOAD_FAST                1 (p)
             90 LOAD_CONST               1 ('expecting regex ')
             92 LOAD_DEREF               3 (model)
             94 FORMAT_VALUE             0
             96 BUILD_STRING             2
             98 CALL                     3

780     >>  106 RETURN_VALUE

None
